v5/core/cr_authenticator.go

```diff
@@ -101,40 +96,87 @@ type ComputeResourceAuthenticator struct {
 
 const (
 	defaultCRTokenFilename = "/var/run/secrets/tokens/vault-token"
-	defaultImdsEndpoint    = "http://169.254.169.254"
-	imdsVersionDate        = "2021-07-15"
-	imdsMetadataFlavor     = "ibm"
-	iamGrantTypeCRToken    = "urn:ibm:params:oauth:grant-type:cr-token" // #nosec G101
-	crtokenLifetime        = 300
+	iamGrantTypeCRToken = "urn:ibm:params:oauth:grant-type:cr-token" // #nosec G101
 )
 
 var craRequestTokenMutex sync.Mutex
 
-// NewComputeResourceAuthenticator constructs a new ComputeResourceAuthenticator instance with the supplied values and
-// invokes the ComputeResourceAuthenticator's Validate() method.
-func NewComputeResourceAuthenticator(crtokenFilename string, instanceMetadataServiceURL, iamProfileName string, iamProfileID string,
```

Comment:
As part of these changes, I decided to go ahead and remote this traditional ctor function in favor of a "builder" instead.